Fetch titles and some description texts from CMS on landing page and search page

nikomakela opened the pull request:
LIIKUNTA-661 LIIKUNTA-755.

Updated HCRC-lib to get new utilities.

Landing page changes:
1. Use hero title as H1 in a search form that is floating on hero element. Old (hard coded production env) translation as default.
2. Use hero description under the hero title. Empty by default.
3. Remove absolute positions, reposition and style the elements related to hero-form-shortcuts -section.
4. Use page content from CMS over shortcuts. It can be used as plain body text or as a description for shortcuts area.
5. Hide the shortcuts and new page content -section on small screens.

Search page changes:
1. Use search page title as H1 in a search form. Old (hard coded production env) translation as default.
2. Render search page content from CMS as text in 1 paragraph under the title.
3. Add a new hard coded text above tabs to describe that section.

nikomakela committed:
chore: upgrade HCRC-lib for new CMS tools

LIIKUNTA-661.

Upgrade HCRC-lib
1. to get page context provider
2. to get getTextFromHtml string util.

nikomakela committed:
feat(sports): search page fetches page heading and descriptions from CMS

LIIKUNTA-661.

nikomakela committed:
feat(sports): landing page fetches title and description from CMS

LIIKUNTA-661 LIIKUNTA-755.

nikomakela committed:
feat(sports): hide landing page shortcuts on small screen

LIIKUNTA-661 LIIKUNTA-755.

Hide the search shortcuts and the CMMS page content section on small
screens.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr889.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
fix(sports): map search view page title and page context

LIIKUNTA-661

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr889.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr889.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela committed:
test: fix landing page browser test

LIIKUNTA-661 LIIKUNTA-755.

The heading is now fetched from CMS, ehich makes it dynamic. The browser
tests cannot search for specific text, because the text can be whatever
is written in CMS content.

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr889.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
refactor: move CmsPageContent -component to shared package-components

LIIKUNTA-661.

With this refactor, also Events and Hobbies can use this same component
more easily.

nikomakela committed:
feat(events): landing page fetches title and description from CMS

LIIKUNTA-661 LIIKUNTA-755.

nikomakela committed:
feat(hobbies): landing page fetches title and description from CMS

LIIKUNTA-661 LIIKUNTA-755.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
feat(events): search page fetches page heading and descriptions from CMS

LIIKUNTA-661.

nikomakela committed:
feat(hobbies): search page fetches page heading and descriptions from CMS

LIIKUNTA-661.

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr889.dev.hel.ninja** :rocket::rocket::rocket:

nikomakela committed:
test: browser test for search cannot test dynamic title

LIIKUNTA-661

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr889.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders reviewed (approved):
Left an observation, fix it or leave it—I wanted to understand `?.` optional chaining operator's handling.

nikomakela committed:
style(sports): set show on map button below search type selector

LIIKUNTA-661.

On mobile sized small screens, the show on map button for venues should
be below the search type selector, not above. It's only shown when venue
type is active, but UI should not change above the selection, but only
below it.

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr889.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr889.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr889.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr889.dev.hel.ninja :laughing::tada::tada::tada:
